ts/Core/Axis/Axis.ts

```diff
@@ -1833,12 +1833,18 @@ class Axis {
         // This is in turn needed in order to find tick positions in ordinal
         // axes.
         if (isXAxis && !secondPass) {
+            // First process all series assigned to that axis.
             axis.series.forEach(function (series): void {
+                // Allows filtering out points outside the plot area.
+                series.forceCrop = series.forceCropping();
+
                 series.processData(
                     axis.min !== (axis.old && axis.old.min) ||
                     axis.max !== (axis.old && axis.old.max)
                 );
             });
+            // Then apply grouping if needed.
+            fireEvent(this, 'afterProcessData');
```

Comment:
The convention in Highcharts is that events with the name `after...` are fired at the end of the function they're named after. So in this case, it either has to have a different name, or the call has to be moved to the end of the `processData` function. 

I would prefer the latter. If the current event handler should only be called in special cases,  we can pass over event arguments to help decide. A potential problem is that StockChart overrides `Series.prototype.processData` so that there is code executing after the parent function. If that is a problem, we can move the whole StockChart processData logic, combined with the new logic, inside an `afterProcessData` event handler.